src/emitter.ts

```diff
@@ -285,18 +285,18 @@ export function emitWebIdl(webidl: Browser.WebIdl, flavor: Flavor, iterator: boo
     function convertDomTypeToTsType(obj: Browser.Typed): string {
         if (obj["override-type"]) return obj["override-type"]!;
         if (!obj.type) throw new Error("Missing type " + JSON.stringify(obj));
-        const type = convertDomTypeToTsTypeWorker(obj);
+        const type = convertDomTypeToTsTypeWorker(obj, false);
         return type.nullable ? makeNullable(type.name) : type.name;
     }
 
-    function convertDomTypeToTsTypeWorker(obj: Browser.Typed): { name: string; nullable: boolean } {
+    function convertDomTypeToTsTypeWorker(obj: Browser.Typed, inUnion: boolean): { name: string; nullable: boolean } {
```

Comment:
Now I need an extra flag to know whether we're converting a type inside of a union type... 🤷

Reply:
I have a suggestion: just use `undefined` inside the converter, and in the method/callback emitter we detect whether the conversion result is a simple `undefined` or `Promise<undefined>` and then replace it as `void` / `Promise<void>`.

This is based on an assumption that those two form are tho only proper uses of `void`.

Reply:
Yay it worked 🎉

Reply:
That also works! 🙂

Even better, applying this change to #890, we will now generate the following for [`ReadableStreamGenericReader`](https://streams.spec.whatwg.org/#readablestreamgenericreader):
```typescript
interface ReadableStreamGenericReader {
    readonly closed: Promise<undefined>;
    cancel(reason?: any): Promise<void>;
}
```
Since `closed` is a property, it keeps its `Promise<undefined>` type, whereas `cancel()` is changed to return `Promise<void>`. I think this is quite neat! 😄